python/repeated.c

```diff
@@ -238,15 +236,14 @@ static PyObject* PyUpb_RepeatedContainer_Item(PyObject* _self,
 PyObject* PyUpb_RepeatedContainer_ToList(PyObject* _self) {
   PyUpb_RepeatedContainer* self = (PyUpb_RepeatedContainer*)_self;
   upb_array* arr = PyUpb_RepeatedContainer_GetIfWritable(self);
-  PyObject* list = PyList_New(0);
-  if (arr) {
-    const upb_fielddef* f = PyUpb_RepeatedContainer_GetField(self);
-    size_t n = upb_array_size(arr);
-    for (size_t i = 0; i < n; i++) {
-      PyObject* val = PyUpb_UpbToPy(upb_array_get(arr, i), f, self->arena);
-      PyList_Append(list, val);
-      Py_DECREF(val);
-    }
+  if (!arr) return PyList_New(0);
+
+  const upb_fielddef* f = PyUpb_RepeatedContainer_GetField(self);
+  size_t n = upb_array_size(arr);
+  PyObject* list = PyList_New(n);
+  for (size_t i = 0; i < n; i++) {
+    PyObject* val = PyUpb_UpbToPy(upb_array_get(arr, i), f, self->arena);
+    PyList_SetItem(list, n, val);
```

Comment:
`SetItem` of `i` I am pretty sure.  This should definitely fail a test somewhere